src/app/clusters/actions-server/actions-server.cpp

```diff
@@ -0,0 +1,462 @@
+        HandleCommand<Commands::EnableAction::DecodableType>(
+            handlerContext, [this](HandlerContext & ctx, const auto & commandData) { HandleEnableAction(ctx, commandData); });
+        return;
+    case Actions::Commands::EnableActionWithDuration::Id:
+        HandleCommand<Commands::EnableActionWithDuration::DecodableType>(
+            handlerContext,
+            [this](HandlerContext & ctx, const auto & commandData) { HandleEnableActionWithDuration(ctx, commandData); });
+        return;
+    case Actions::Commands::DisableAction::Id:
+        HandleCommand<Commands::DisableAction::DecodableType>(
+            handlerContext, [this](HandlerContext & ctx, const auto & commandData) { HandleDisableAction(ctx, commandData); });
+        return;
+    case Actions::Commands::DisableActionWithDuration::Id:
+        HandleCommand<Commands::DisableActionWithDuration::DecodableType>(
+            handlerContext,
+            [this](HandlerContext & ctx, const auto & commandData) { HandleDisableActionWithDuration(ctx, commandData); });
+        return;
+    }
+}
+
+CHIP_ERROR ActionsServer::ModifyActionList(EndpointId aEndpoint, const ActionStructStorage & aAction)
+{
+    Delegate * delegate = GetDelegate(aEndpoint);
+    VerifyOrReturnError(IsValid(delegate) == CHIP_NO_ERROR, CHIP_ERROR_INCORRECT_STATE);
+
+    // Read through the list to find and update the existing action that matches the passed-in action's ID.
+    for (uint16_t i = 0; i < kMaxActionListLength; i++)
+    {
+        ActionStructStorage existingAction;
+        CHIP_ERROR err = delegate->ReadActionAtIndex(i, existingAction);
+        if (err == CHIP_ERROR_PROVIDER_LIST_EXHAUSTED)
+        {
+            break;
+        }
+
+        if (existingAction.actionID == aAction.actionID)
+        {
+            existingAction.Set(aAction.actionID, aAction.name, aAction.type, aAction.endpointListID, aAction.supportedCommands,
+                               aAction.state);
+
+            MarkDirty(aEndpoint, Attributes::ActionList::Id);
+            return CHIP_NO_ERROR;
+        }
+    }
+
+    return CHIP_ERROR_NOT_FOUND;
+}
+
+CHIP_ERROR ActionsServer::ModifyEndpointList(EndpointId aEndpoint, const EndpointListStorage & aEpList)
+{
+    Delegate * delegate = GetDelegate(aEndpoint);
+    VerifyOrReturnError(IsValid(delegate) == CHIP_NO_ERROR, CHIP_ERROR_INCORRECT_STATE);
+
+    // Read through the list to find and update the existing action that matches the passed-in endpoint-list's ID
+    for (uint16_t i = 0; i < kMaxEndpointListLength; i++)
+    {
+        EndpointListStorage existingEpList;
+        CHIP_ERROR err = delegate->ReadEndpointListAtIndex(i, existingEpList);
+        if (err == CHIP_ERROR_PROVIDER_LIST_EXHAUSTED)
+        {
+            break;
+        }
+
+        if (existingEpList.endpointListID == aEpList.endpointListID)
+        {
+            existingEpList.Set(aEpList.endpointListID, aEpList.name, aEpList.type, aEpList.endpoints);
```

Comment:
Again, this is changing something on the stack that is then no longer used.  What's the point?

Reply:
Removed this code, as this did not make any sense.